pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -290,6 +291,38 @@ func validatePipelineContextVariables(tasks []PipelineTask) *apis.FieldError {
 	return errs.Also(validatePipelineContextVariablesInParamValues(paramValues, "context\\.pipeline", pipelineContextNames))
 }
 
+func validateExecutionStatusVariables(tasks []PipelineTask, finallyTasks []PipelineTask) (errs *apis.FieldError) {
+	// creating a list of pipelineTask names to validate tasks.<name>.status
+	pipelineRunTasksContextNames := sets.String{}
+	for idx, t := range tasks {
+		for _, param := range t.Params {
+			// validate dag pipeline tasks not accessing execution status of other pipeline task
+			if ps, ok := GetVarSubstitutionExpressionsForParam(param); ok {
+				for _, p := range ps {
+					if strings.HasPrefix(p, "tasks") && strings.HasSuffix(p, "status") {
```

Comment:
I suggest tightening this check up a bit by including the periods.

`if strings.HasPrefix(p, "tasks.") && strings.HasSuffix(p, ".status") {`

Reply:
Done 👍